### PR TITLE
AER-2445 refactored out GSON library

### PR DESCRIPTION
**Pull request by Hilbrand**
Also let taskmanager-parent extend from aerius root pom.

### CODE REVIEW

File: source/taskmanager/src/main/java/nl/aerius/taskmanager/domain/PriorityTaskQueue.java [hunk 1]
```diff
@@ -43,11 +34,10 @@ public PriorityTaskQueue() {
 
   /**
    * @param queueName The name of the queue the task corresponds to.
-   * @param description The description of this task queue.
    * @param priority The priority this task should have.
    * @param maxCapacityUse The maximum capacity this task can use of the total workers assigned. Should be a fraction.
    */
-  public PriorityTaskQueue(final String queueName, final String description, final int priority, final double maxCapacityUse) {
+  public PriorityTaskQueue(final String queueName,final int priority, final double maxCapacityUse) {
```
Comment (BertScholten) [inline source/taskmanager/src/main/java/nl/aerius/taskmanager/domain/PriorityTaskQueue.java:40]:
```suggestion
  public PriorityTaskQueue(final String queueName, final int priority, final double maxCapacityUse) {
```

File: source/taskmanager/src/main/java/nl/aerius/taskmanager/PriorityTaskSchedulerFileHandler.java [hunk 2]
```diff
@@ -53,26 +49,16 @@ public PriorityTaskSchedule read(final File file) throws IOException {
...
 
   private PriorityTaskSchedule readFromFile(final File file) throws IOException {
-    try (final Reader reader = Files.newBufferedReader(file.toPath(), StandardCharsets.UTF_8)) {
-      return gson.fromJson(reader, PriorityTaskSchedule.class);
-    }
+    return objectMapper.readValue(file, PriorityTaskSchedule.class);
   }
 
-  private PriorityTaskSchedule readFromEnvironment(final String workerQueueName) {
+  private PriorityTaskSchedule readFromEnvironment(final String workerQueueName) throws JsonMappingException, JsonProcessingException {
```
Comment (BertScholten) [inline source/taskmanager/src/main/java/nl/aerius/taskmanager/PriorityTaskSchedulerFileHandler.java:55]:
```suggestion
  private PriorityTaskSchedule readFromEnvironment(final String workerQueueName) throws IOException {
```
Think these are subclasses of `IOException` at least.

**Review by BertScholten — APPROVED** [review]
LGTM, few suggestions

File: source/taskmanager/src/main/java/nl/aerius/taskmanager/PriorityTaskSchedulerFileHandler.java (hunk 2 above)
Comment (Hilbrand) [reply source/taskmanager/src/main/java/nl/aerius/taskmanager/PriorityTaskSchedulerFileHandler.java:55]: Good point. Turns out `JsonMappingException` is a subclass of `JsonProcessingException`. Therefore I can reduce this to `JsonProcessingException`. They are indeed subclasses of `IOException`. But with only 1 exception left I keep that one.

**Pull request merged**